### PR TITLE
Fixup my bad volume type mismatch PR

### PR DESCRIPTION
**Pull request by garrettwrong**
Whoops, I put the fix in simulation, but didn't commit the function in volume...  Added a test as penance.

### CODE REVIEW

**Comment by garrettwrong** [comment]
OSX CI fail is a brownout, will create issue.

File: tests/test_volume.py [hunk 1]
```diff
@@ -40,6 +40,18 @@ def tearDown(self):
...
+        if self.dtype == np.float64:
+            new_dtype = np.float32
+        elif self.dtype == np.float32:
+            new_dtype = np.float64
+        else:
+            skip("Skip numerically comparing non float types.")
+
+        v2 = self.vols_1.astype(new_dtype)
+        self.assertTrue(isinstance(v2, Volume))
+        self.assertTrue(np.allclose(v2.asnumpy(), self.vols_1.asnumpy()))
```
Comment (j-c-c) [inline tests/test_volume.py:53]: Should we be checking that `vs.asnumpy().dtype == new_dtype`?

**Review by j-c-c — COMMENTED** [review]
Looks good. One small question?

File: tests/test_volume.py (hunk 1 above)
Comment (garrettwrong) [reply tests/test_volume.py:53]: Sure, added line.

**Review by j-c-c — APPROVED** [review]
Looks good.

**Comment by garrettwrong** [comment]
I think we can merge this little guy.

**Pull request merged**